### ISSUE TITLE
SVG icons to replace htmlentities in right buttons

### ISSUE THREAD

**Issue by jnvsor** [kint-php/kint#432]
@DRSDavidSoft I could handle this if you're too busy, no rush anyway

**Comment by jnvsor** [comment]
![image](https://github.com/user-attachments/assets/da51b9ba-3cb0-4b61-b486-e747ab72d532)

WIP - any comments? @DRSDavidSoft ?

**Comment by DRSDavidSoft** [comment]
Thank you for creating these, they look great! 👍🏻 As a WIP they look amazing for the starting point!

Personally, I only started working on the 🔎 icon, before work got busy, but the resulting design wasn't to my liking, so I haven't yet submitted it.  I'm unfortunately still busy at work, for just a little more, but I'll be back soon to continue working on the open issues. In the meantime, I'd appreciate it if you could either attach the SVG files themselves, here or even better, create a new branch that would contains the assets, that way I could also submit PRs directly towards that branch.

Once again, thank you, the attention to detail while re-designing the interface is certainly appreciated ❤️

**Comment by jnvsor** [comment]
Do you mind if I just stick these in master and let you revamp them later? I want to release a 6.0.1 soon since I've got quite a few minor bugfixes waiting for a tag.

(Lazy object support will have to wait till 6.1 since it's a bit more complex than I thought)

Side-note: Do you know if there's a simple way to import SVGs into sass? We're approaching double digits inlined svg files and it would probably be cleaner long-term to import them automatically, but I'd like to avoid a full-fledged JS build system

![access_path](https://github.com/user-attachments/assets/ad02ef26-d0c5-4c35-b35e-24f4db42fab7)
![folder](https://github.com/user-attachments/assets/1d60d3a2-d267-4ec2-958f-a73f0e3eba07)
![search](https://github.com/user-attachments/assets/adc18dfe-f4ac-482f-aff1-ee0247fbafe4)

**Comment by DRSDavidSoft** [comment]
> Do you mind if I just stick these in master and let you revamp them later? 

I don't mind, but is there a particular reason not to create a branch specifically for this? I assume directly committing to master would be more convenient.

> Do you know if there's a simple way to import SVGs into sass? We're approaching double digits inlined svg files and it would probably be cleaner long-term to import them automatically, but I'd like to avoid a full-fledged JS build system

There are some small npm packages that are especially designed for this. I can work on a PR for it if you accept this route. Also, personally I wouldn't mind a full-fledged build system for the client side, if it'd be kept simple and fast.

Also, thanks for attaching the SVGs, I'll try to make sure the new icons would be included in 6.0.1. Any upgrades and improvements can make it in 6.1! As always, thanks for all the hard work you do, each new release is always exciting!

**Comment by jnvsor** [comment]
> I don't mind, but is there a particular reason not to create a branch specifically for this?

I was recently made aware of how bad it looks on other operating systems (Chiefly windows) so I really want a bear minimum released asap - we can make a branch to work on but a PR branch would be fine too

> I can work on a PR for it if you accept this route.

Sounds great. Should I just put these in for now and open a new issue for JS build tweaks or do you want to tidy them up first?

**Comment by DRSDavidSoft** [comment]
> how bad it looks on other operating systems (Chiefly windows)

If it's not too off-topic, can you please elaborate, are you referring to how bad the current icons look on Windows, or something else?

> Should I just put these in for now and open a new issue for JS build tweaks or do you want to tidy them up first?

Let's commit the current icons and open an issue for the build system, I'll address both separately

I'm a bit low on time so I'd like to apologize if any of the tasks take longer, I'll try to submit a simple solution and apply any improvements/enhancements later down the line

Thank you

**Comment by jnvsor** [comment]
> are you referring to how bad the current icons look on Windows

Yeah exactly. I assumed if they had htmlentities they'd look good on all systems or at least on systems with the right fonts.

I'll try to get these icons in but now I'm running into a new problem: When inserted as a background image you can't use CSS to set the colors and I don't want to bloat the output with a bunch of SVG markup. I'll probably end up using the new css mask property along with ::content

**Comment by jnvsor** [comment]
@DRSDavidSoft The best laid plans...

They look simple enough like that but needed to be much simpler to fit in the UI. The access path one was particularly illegible but I came up with a solution that's also much better since you can tell what it does just by looking at it.

Chrome/FF at normal zoom on a normal DPI monitor:

![image](https://github.com/user-attachments/assets/18cb8fc2-c3fd-4a5d-996c-0d8441c86f6e)

As you can see there's so little to work with I even had to get rid of the arrow in the folder...

I'll upload the source files below. If you have no complaints I'll push this and tag it 6.0.1
![access_path](https://github.com/user-attachments/assets/94a1c5de-c0d5-432c-b39f-7aec189b7284)
![folder](https://github.com/user-attachments/assets/fa4d3089-a77c-4ea8-b403-9d10e9960312)
![search](https://github.com/user-attachments/assets/68e24a7a-8194-43d2-ab4a-7c024239adf4)

**Comment by jnvsor** [comment]
Merged in 2fb169bfad395077cb5bf3d00c5c8556ffe66f3d and tagged 6.0.1

I'll leave this open for later discussion @DRSDavidSoft
